railties/lib/rails/generators/rails/app/templates/config/puma.rb.tt

```diff
@@ -17,9 +26,8 @@ worker_timeout 3600 if ENV.fetch("RAILS_ENV", "development") == "development"
 #
 port ENV.fetch("PORT") { 3000 }
 
-# Specifies the `environment` that Puma will run in.
-#
-environment ENV.fetch("RAILS_ENV") { "development" }
+# Specifies the Rack `environment` that Puma will run in.
+environment ENV.fetch('RACK_ENV') { 'development' }
```

Comment:
Would not this mean that `RAILS_ENV=production rails server` will start the puma environment to development?

Reply:
Would by default unless 'RACK_ENV' is set to production, which it is on Heroku: https://devcenter.heroku.com/articles/deploying-rails-applications-with-the-puma-web-server#environment

Reply:
@rafaelfranca we can have the default fallback to the RAILS_ENV if RACK_ENV is not defined.

```ruby
environment ENV.fetch('RACK_ENV') { ENV.fetch["RAILS_ENV"] }
```